src/software/world/team.h

```diff
@@ -119,6 +119,17 @@ class Team
      */
     void setRobotExpiryBuffer(const Duration& new_robot_expiry_buffer_duration);
 
+    /**
+     * Updates the unavailable capabilities of a specific robot. If the team does
+     * not have that robot, does not do anything.
+     *
+     * @param id the id of the desired Robot
+     * @param new_unavailable_robot_capabilities a set of the updated unavailable
+     * robot capabilities
+     * */
+    void setUnavailableRobotCapabilities(
```

Comment:
I'm curious why this function was added to `Team`, rather than, for example, adding a `getMutableRobots` function and using that to change the capabilities.

Reply:
Both of them would work. It was more of a design choice to not have a `getMutableRobots` function since it would give a lot of power to the user. I was talking to @jonathanlew about this, he might have some more to pitch in.

Reply:
yes that's it, it's about encapsulation